File: PAT_Basic_Level_by_lua/1081 检查密码/1081 检查密码.lua
-- 题目编号: 1081 检查密码
-- 实现原理: 逐行读取密码字符串，分别检查长度是否>=6、是否只包含字母数字和点、是否包含字母、是否包含数字，按缺失情况输出对应提示信息。
local n = tonumber(io.read())  -- 读取密码数量

-- 逐个检查每个密码
for i = 1, n do
    local s = io.read("l")  -- 读取密码字符串
    local l = #s  -- 密码长度
    local a = s:match("%a")  -- 检查是否包含字母（%a匹配字母）
    local d = s:match("%d")  -- 检查是否包含数字（%d匹配数字）
    local ok = s:match("^[%w%.]+$")  -- 检查是否只包含字母数字和点（%w匹配字母数字，%.匹配点号）
    
    -- 按优先级判断并输出对应提示
    if l < 6 then
        -- 长度不足6位
        print("Your password is tai duan le.")
    elseif not ok then
        -- 包含非法字符
        print("Your password is tai luan le.")
    elseif not a then
        -- 缺少字母
        print("Your password needs zi mu.")
    elseif not d then
        -- 缺少数字
        print("Your password needs shu zi.")
    else
        -- 全部符合要求
        print("Your password is wan mei.")
    end
end

File: PAT_Basic_Level_by_c/1081 检查密码/1081 检查密码.c
/*
 * 题目：1081 检查密码
 * 实现原理：
 *   检查密码是否符合要求，按优先级输出相应提示：
 *   1. 长度不足6位："Your password is tai duan le."
 *   2. 包含非法字符（非字母、非数字、非点号）："Your password is tai luan le."
 *   3. 缺少数字："Your password needs shu zi."
 *   4. 缺少字母："Your password needs zi mu."
 *   5. 全部符合要求："Your password is wan mei."
 *   算法步骤：
 *   1. 读取密码数量n
 *   2. 逐行读取密码，统计长度和字符类型
 *   3. 按优先级判断并输出相应提示
 */
#include <stdio.h>
#include <ctype.h>

int main() {
    int n;
    scanf("%d", &n);
    getchar();              // 吃掉第一行末尾的换行符
    for (int i = 0; i < n; i++) {
        char c;
        int len = 0, has_num = 0, has_alpha = 0, has_invalid = 0;   // 分别记录长度、有无数字/字母/非法字符
        while ((c = getchar()) != '\n') {       // 逐字符读入一行密码（密码可含空格）
            len++;
            if (isdigit(c)) {
                has_num = 1;
            } else if (isalpha(c)) {
                has_alpha = 1;
            } else if (c != '.') {              // 除字母、数字、点号外均为非法字符
                has_invalid = 1;
            }
        }
        if (len < 6) {                          // 按优先级依次判断并输出提示
            printf("Your password is tai duan le.\n");
        } else if (has_invalid) {
            printf("Your password is tai luan le.\n");
        } else if (!has_num) {
            printf("Your password needs shu zi.\n");
        } else if (!has_alpha) {
            printf("Your password needs zi mu.\n");
        } else {
            printf("Your password is wan mei.\n");
        }
    }
    return 0;
}